server/src-lib/Hasura/GraphQL/Execute/Action.hs

```diff
@@ -619,23 +532,15 @@ mkJsonAggSelect =
   bool RS.JASSingleObject RS.JASMultipleRows . isListType
 
 processOutputSelectionSet
-  :: ( MonadReusability m
-     , MonadError QErr m
-     , MonadReader r m
-     , Has FieldMap r
-     , Has OrdByCtx r
-     , Has SQLGenCtx r
-     )
-  => RS.ArgumentExp UnresolvedVal
+  :: RS.ArgumentExp v
```

Comment:
```suggestion
  :: RS.ArgumentExp UnpreparedValue
```

Reply:
What is the advantage of using a more specific type in this case?

Reply:
Perhaps, better type-safety?

Reply:
We gain nothing by making this function more specific: the fact that it works for any type `v` without any restriction means that the actual data stored in it isn't used in any way; it is purely a "structural" transformation. As such, why should we restrict it to a specific kind of data? We also have better type safety that way, since the compiler will enforce we're not doing anything with the data.

Reply:
Oh yes, you folks are right. It's better to have `RS.ArgumentExp v` .